MediaToolkit src/MediaToolkit/Engine.cs

```diff
@@ -291,6 +307,25 @@ private void StartFFmpegProcess(EngineParameters engineParameters)
                     }
                 };
 
+                if(engineParameters.ReturningStandardOutput)
+                {
+                    FileStream baseStream = this.FFmpegProcess.StandardOutput.BaseStream as FileStream;
+                    this.thumbnailByteArray = null;
+                    int lastRead = 0;
+
+                    using (MemoryStream ms = new MemoryStream())
+                    {
+                        byte[] buffer = new byte[4096];
+                        do
+                        {
+                            lastRead = baseStream.Read(buffer, 0, buffer.Length);
+                            ms.Write(buffer, 0, lastRead);
+                        } while (lastRead > 0);
+
+                        this.thumbnailByteArray = ms.ToArray();
+                    }
```

Comment:
Why not try reading from `baseStream` all in one go?
```
byte[] buffer = new byte[baseStream.Length];
baseStream.Read(buffer, 0, buffer.Length);
this.thumbnailByteArray = buffer;
```
Only reason I could think as to why it wouldn't work is if `baseStream.CanSeek` is false.